components/brave_perf_predictor/browser/bandwidth_linreg.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_BANDWIDTH_LINREG_H_
+#define BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_BANDWIDTH_LINREG_H_
+
+#include <string>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "brave/components/brave_perf_predictor/browser/bandwidth_linreg_parameters.h"
+
+namespace brave_perf_predictor {
+
+// Computes prediction based on the provided feature vector.
+// It is the client's responsibility to provide features in
+// the exact order expected by the predictor.
+double LinregPredictVector(const std::array<double, feature_count>& features);
```

Comment:
I would rather see what the method actually predicts in its name (`PredictSavingsBytesFromVector` ?), but not a problem

Reply:
this is an implementation detail of `bandwidth_savings_predictor` - linear regression has no notion of "bytes" or "savings", it collects provided feature vector, computes vector product with predefined features and returns a number. Linreg here is significant because it tells what is done to the provided feature vector

Reply:
makes sense